### PR TITLE
Shorten GPG keypath for deb packaging

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we apply the same GPG path shortening logic for
the RPM and the DEB packaging modules.

Relates #17053

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
@nik9000 Could you please check?

File: distribution/correct-sign-path.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            <and>
+                <isfalse value="${gpg.keypath.overridden}"/>
+                <os family="windows"/>
+            </and>
+        </condition>
+
+    </target>
+
+    <!--
+        Either use a symlink (Unix) to shorten the GPG path or copy the directory (Windows).
```
Comment (nik9000) [inline distribution/correct-sign-path.xml]: I don't think we even try to sign these on windows. Its worth testing there, but I think we skip the whole thing.

Honestly I'd be totally fine skipping deb and rpm entirely on windows.

File: distribution/pom.xml [hunk 2]
```diff
@@ -45,6 +45,7 @@
         <!-- we expect packaging formats to have integration tests, but not unit tests -->
         <skip.unit.tests>true</skip.unit.tests>
 
+        <packaging.gpg.shortening.ant.script.dir>${project.basedir}/../correct-sign-path.xml</packaging.gpg.shortening.ant.script.dir>
```
Comment (nik9000) [inline distribution/pom.xml]: remove the .dir from the end of this name I think.

File: distribution/correct-sign-path.xml (hunk 1 above)
Comment (danielmitterdorfer) [inline distribution/correct-sign-path.xml]: We do sign on Windows (at least the deb plugin checks the path name). I can create a separate issue for skipping the package builds on Windows.

File: distribution/pom.xml (hunk 2 above)
Comment (danielmitterdorfer) [inline distribution/pom.xml]: Makes sense. I'll correct that.

**Pull request merged**